### PR TITLE
Fix #9054: remove gSceneryTileElement by returning it within GameActi…

### LINKED ISSUE #9054: Remove gSceneryTileElement by returning it within GameActionResult
Another similar one to #9053 but this time for Wall, Large and Small Scenery Place Actions. Will require returning a subclass of GameActionResult with the correct additional field.

This will help reduce number of global vars that litter the codebase.

Note this cannot be completed until LargeSceneryPlaceAction is finished #9055.

### PR DESCRIPTION
**Pull request by boucks**
Removes gSceneryTileElement as per #9054. Let me know if there should be any other changes.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Just one or two clang format issues you will need to adjust. Other than that looks great.

File: src/openrct2/actions/WallPlaceAction.hpp [hunk 1]
```diff
@@ -364,14 +391,14 @@ DEFINE_GAME_ACTION(WallPlaceAction, GAME_COMMAND_PLACE_WALL, GameActionResult)
...
-                    GA_ERROR::NO_CLEARANCE, STR_CANT_BUILD_PARK_ENTRANCE_HERE, gGameCommandErrorText, gCommonFormatArgs);
+                return std::make_unique<WallPlaceActionResult>(
+                    GA_ERROR::NO_CLEARANCE, gGameCommandErrorText, gCommonFormatArgs);
             }
         }
 
         if (!map_check_free_elements_and_reorganise(1))
         {
-            return MakeResult(GA_ERROR::NO_FREE_ELEMENTS, STR_CANT_BUILD_PARK_ENTRANCE_HERE, gGameCommandErrorText);
+            return std::make_unique<WallPlaceActionResult>(GA_ERROR::NO_FREE_ELEMENTS, gGameCommandErrorText);
```
Comment (ZehMatt) [inline src/openrct2/actions/WallPlaceAction.hpp]: MakeResult already returns the correct version, why was this changed? MakeResult should be used in all those cases.

File: src/openrct2/actions/WallPlaceAction.hpp [hunk 2]
```diff
@@ -23,7 +23,34 @@
...
+        : GameActionResult(err, STR_CANT_BUILD_PARK_ENTRANCE_HERE, msg)
+    {
+    }
+
+    WallPlaceActionResult(GA_ERROR error, rct_string_id msg, uint8_t* args)
+        : GameActionResult(error, STR_CANT_BUILD_PARK_ENTRANCE_HERE, msg, args)
+    {
+    }
+
+    TileElement* tileElement;
```
Comment (ZehMatt) [inline src/openrct2/actions/WallPlaceAction.hpp]: Uninitialized variable.

File: src/openrct2/actions/SmallSceneryPlaceAction.hpp [hunk 3]
```diff
@@ -48,6 +48,7 @@ class SmallSceneryPlaceActionResult final : public GameActionResult
     }
 
     uint8_t GroundFlags{ 0 };
+    TileElement* tileElement;
```
Comment (ZehMatt) [inline src/openrct2/actions/SmallSceneryPlaceAction.hpp]: Uninitialized variable.

File: src/openrct2/actions/LargeSceneryPlaceAction.hpp [hunk 4]
```diff
@@ -40,6 +40,7 @@ class LargeSceneryPlaceActionResult final : public GameActionResult
     }
 
     uint8_t GroundFlags{ 0 };
+    TileElement* tileElement;
```
Comment (ZehMatt) [inline src/openrct2/actions/LargeSceneryPlaceAction.hpp]: Uninitialized variable.

File: src/openrct2/actions/WallPlaceAction.hpp (hunk 1 above)
Comment (boucks) [reply src/openrct2/actions/WallPlaceAction.hpp]: I was following along to the process of #8726 which was similar in this issue, MakeResult needs to be updated to call the constructor of my Subclass in order to fix the issue in #9054 , unless I am mistaken

**Comment by Broxzier** [comment]
Are you making these format changes manually? clang-format can format the code for you.

**Comment by boucks** [comment]
Yeah I was, sorry for making so many extra commits. I'm not sure how to run the clang-format. It seemed that somethings changed automatically for me in VS but these changes did not.

File: src/openrct2/actions/WallPlaceAction.hpp (hunk 1 above)
Comment (ZehMatt) [reply src/openrct2/actions/WallPlaceAction.hpp]: It should already use your subclass if you specified it via DEFINE_GAME_ACTION, it will use the template parameter.

Comment (boucks) [reply src/openrct2/actions/WallPlaceAction.hpp]: @duncanspumpkin Should I revert back to using MakeResult?

Comment (duncanspumpkin) [reply src/openrct2/actions/WallPlaceAction.hpp]: I wasn't overly fussed which way it was done since there isn't consistency across the other game actions right now. It can either be reverted to make result or we can catch all the game actions in a make result pr

Comment (ZehMatt) [reply src/openrct2/actions/WallPlaceAction.hpp]: Works for me, however I just pointed it out as those changes are somewhat unnecessary given it already has MakeResult, but fine.

File: src/openrct2/actions/WallPlaceAction.hpp (hunk 2 above)
Comment (boucks) [reply src/openrct2/actions/WallPlaceAction.hpp]: How should I resolve this, should I call the default TileElement constructor?

**Comment by Nazey** [comment]
@boucks if you hit edit, then advance, it should be format document. It will use the clang-format file found in the ORCT2 folder.

File: src/openrct2/actions/WallPlaceAction.hpp (hunk 2 above)
Comment (duncanspumpkin) [reply src/openrct2/actions/WallPlaceAction.hpp]: Default value of nullptr will be fine

Comment (boucks) [reply src/openrct2/actions/WallPlaceAction.hpp]: Gotcha, committing now.
